packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue

```diff
@@ -0,0 +1,215 @@
+          />
+        </div>
+        <div class="btn-container">
+          <input
+            type="button"
+            class="btn red-button"
+            @click="!isDisabledSocialAuth && setTaCUrlAndRedirect()"
+            :class="{ 'btn-disabled': isDisabledSocialAuth }"
+            value="continue"
+            :disabled="isDisabledSocialAuth"
+          />
+        </div>
+      </div>
+    </template>
+  </DialogWindow>
+</template>
+
+<script>
+import buildUrl from "build-url";
+import {
+  countriesList,
+  americanStates,
+  canadianProvinces,
+  GDPR_COUNTRIES
+} from "../const/signup.const";
+
+const CANADA = "Canada";
+const USA = "United States";
+const EMPTY_STRING = "";
+
+export default {
+  name: "TermsAndConditionsDialog",
+  components: {
+    SmartLink: () => import("./SmartLink"),
+    DialogWindow: () => import("./DialogWindow")
+  },
+  props: {
+    socialUrl: {
+      required: true,
+      type: EMPTY_STRING
+    }
+  },
+  data() {
+    return {
+      selectedCountry: EMPTY_STRING,
+      selectedRegion: EMPTY_STRING,
+      regionData: { type: EMPTY_STRING, list: [] }
+    };
+  },
+  computed: {
+    isDisabledSocialAuth() {
+      // If country not selected, disable "continue" button.
+      if (!this.selectedCountry) {
+        return true;
+      }
+      // If country Canada or USA but region not selected, disable "continue" button.
+      if (this.selectedCountry === USA || this.selectedCountry === CANADA) {
+        if (!this.selectedRegion) {
+          return true;
+        }
+      }
+    },
+    region: {
+      get() {
+        return this.regionData;
+      },
+      set(country) {
+        if (country === USA) {
+          this.regionData.list = americanStates;
+          this.regionData.type = "State";
+        } else if (country === CANADA) {
+          this.regionData.list = canadianProvinces;
+          this.regionData.type = "Province";
+        } else {
+          this.regionData.list = [];
+          this.regionData.type = EMPTY_STRING;
+        }
+      }
+    },
+    countriesList() {
+      return countriesList;
+    },
+    isGDPRCountry() {
+      if (GDPR_COUNTRIES.includes(this.selectedCountry)) {
+        return true;
+      }
+    }
+  },
+  methods: {
+    setTaCUrlAndRedirect() {
+      const okta_AcceptedToS = this.$refs.gdprBoxEl.checked;
+      const okta_ts_AcceptedToS = Date.now();
+      const country = this.selectedCountry;
+      const regionType = this.region.type.toLowerCase(); // could be "state" or "province".
+      const regionName = this.selectedRegion;
+      const queryParams = this.removeEmptyFields({
+        okta_AcceptedToS,
+        okta_ts_AcceptedToS,
+        country,
+        [regionType]: regionName
+      });
+      
+      window.location.href = buildUrl(this.socialUrl, { queryParams });
+    },
+    closeTermsConditionsDialog() {
+      this.$emit("close");
+    },
+
+    // Remove object fields with empty "key" value.
```

Comment:
How can the object have fields with an empty key value? Based on the `queryParams` definition above the only possible one I can see is the dynamically inferred `[regionType]` key.

Reply:
yes in this case it's only one property that could be empty but logic could be changed. e.g. a new property added and in some cases it could be empty

Reply:
By definition only one key could be empty though 🤔

Reply:
I'm not sure that I've caught your thought, could you please describe in more detail?

Reply:
The comment says "Remove object fields with empty "key" value." But hash key/value pairs mean only only one entry with any "key", including `""` e.g.
```js
const myObj = {};
myObj[""] = "some value";

console.log(myObj); // { '': 'some value' }

myObj[""] = "other value";

console.log(myObj); // { '': 'other value' }
```
Only one entry can have any particular key ^

Reply:
well spotted! I haven't thought about that, thanks.
 what if to use `delete obj[""]` ?

Reply:
Using `delete` would work -- I think overall it might be clearer to directly check if `regionType` is not empty, then add it to `queryParams` like this:
```js
const queryParams = {
        okta_AcceptedToS,
        okta_ts_AcceptedToS,
        country,
      };
if (regionType !== "") {
  queryParams[regionType] = regionName;
}
```